.github/workflows/fmt.yml

```diff
@@ -0,0 +1,32 @@
+name: fmt
```

Comment:
Does it make sense to run this check against stable, beta and nightly?

Is there any chance that this will fail for one version and not for the others? If that happen how will we fix that?

Reply:
Keeps our codebase open for `beta` and `nightly` users as well.

Some people rather use `nightly` as their channel.

This is a _good to have_ when you are working on  a library crate for instance, but i don't think it hurts for our use case to be honest.

As we are building a binary crate, this help us making sure future releases are going to still support what we have.

Also if we are likely to implement a feature which relies on using some future version feature (nightly, beta), we can work on it and have checks on them until that PR with the feature passes all the checks. You could also have a bot merging PRs like these automatically if that's the case.

Reply:
I mean, I think that it is fine to run cargo build, cargo test and cargo clippy against those channels. But I think that cargo fmt should run only on stable